### PR TITLE
Making ambiguities report as errors

### PR DESCRIPTION
**Pull request by radumereuta**
also adding --keep-amb for debugging purposes

This should be a fix for #828

### CODE REVIEW

**Comment by radumereuta** [comment]
@dwightguth please review.

File: k-distribution/tests/regression-new/context-alias/test.k [hunk 1]
```diff
@@ -10,12 +10,12 @@ context alias [heatable]: <k> HERE ...</k> <stuff> true </stuff>
 
 context alias [heatable2]: <k> HERE ...</k> <shouldHeat> B </shouldHeat> requires B
 
-syntax Expr ::= Expr "+" Expr [strict(heatable)]
```
Comment (dwightguth) [inline k-distribution/tests/regression-new/context-alias/test.k:13]: I'd like us to wait to merge this PR until my PR fixing this particular set of tests in a better way is merged.

**Review by dwightguth — CHANGES_REQUESTED** [review]
I haven't looked over this in detail, but let's remove the --keep-amb flag. You can make a separate PR for that if you want, but I'd like to keep this PR focused on reporting ambiguities as errors.

File: k-distribution/tests/regression-new/context-alias/test.k (hunk 1 above)
Comment (radumereuta) [reply k-distribution/tests/regression-new/context-alias/test.k:13]: Let me know when that's done and I'll come back to this.

**Pull request closed without merging**